### DISCUSSION TITLE
Adding screw-together baseplate functionality

### DISCUSSION THREAD

**Discussion by jazzhands** [kennetek/gridfinity-rebuilt-openscad#40] [General]
Hi all, I created a pull request to add screw-together baseplate functionality ( based on Kyle Warren's work in Fusion 360: https://www.printables.com/model/300603-gridfinity-screw-together-baseplate ) to this excellent project (thanks kennetek, this is a wonderful thing you started).

I wanted to talk about it because:

1. adding this functionality added several new things to the customizer to describe the kinds of screws used. In reality, I don't think most users will change the defaults (they're set to be pretty good for M3 and 4-40 size screws, common metric and imperial sizes).  Should I move those variables out of the customizer and let advanced users adjust in code if they want?
2. Unlike Kyle Warren, I think 3 screws per square is excessive. I left the default at 1 hole per grid, but I let the customizer add 1,2,or 3 holes per grid if the user wishes. 1 and 3 holes per grid will be compatible, but if you choose 2 holes per grid, those baseplates won't work with 1 or 3. Should I just remove the option for 2 holes?
3. I haven't checked to see if these are compatible with Kyle Warren's baseplates. I chose a Z height that worked with kennetek's skeletonized model, I didn't try to match Kyle Warren's baseplates. I left the holes relatively large to allow for a bit of clearance and differences between printers and models.
4. I could use some more testers (I'm just one person with one 3D printer) - horizontal holes in FDM 3D printed parts aren't the greatest idea in the world, they get squished and lose their dimensional accuracy. If that's a real problem, I can easily change the holes to hexagons instead of circles.

**Comment by timmgleason** [comment]
I was coming here to actually request this same thing.
Got the post written, then the checkbox 'search for similar' caught my attention, and well here I am.

Was your PR rejected? I do not see it in the PR tab. I see you've committed code, but a search of this repo for your code came up empty. Could be me, I do not use Github at work and the search function on Bitbucket in our implementation is broken as well.

Let me know!

**Comment by timmgleason** [comment]
Ah, found the code in the current HEAD of gridfinity-rebuilt-utility.scad, but then found I needed the standard,scad library also.

Tinkering with it now.

**Reply by Ruudjhuu** [reply]
Glad you found it. Currently this repo does not have dependencies. You can use the baseplates file, open it and choose for one of the screw together styles

**Reply by timmgleason** [reply]
While I did not clone the repo, just dld the file ([56ca5e4](https://github.com/kennetek/gridfinity-rebuilt-openscad/commit/56ca5e48619eed3cfd4788001b3bcd1e5f814409)), gridfinity-rebuilt-baseplate.scad currently has an include statement for standard.scad at the top of it.

Also found that the magnet holes to not render in the correct places, and only down the center of the plate.
![image](https://github.com/kennetek/gridfinity-rebuilt-openscad/assets/11194213/53b2a003-266e-4db7-814b-a7cfc90208e0)

Once I learn more (nay anything) about the language, I'd be happy to contribute.

**Reply by timmgleason** [reply]
Although I now suspect it is because I have the incorrect gridfinity-rebuilt-utility.scad file.

**Reply by timmgleason** [reply]
Yes, that was the problem.
After cloning the repo, everything is as it should be.
Although the baseplate file does indeed have an include of the standard.scad file
